test/libsolidity/ViewPureChecker.cpp

```diff
@@ -128,17 +128,21 @@ BOOST_AUTO_TEST_CASE(environment_access)
 	};
 	for (string const& x: view)
 	{
-		CHECK_ERROR(
+	    // Warning: Use of var keyword is deprecated
+	    // Error (TypeError):
+		//	Function declared as pure, but this expression (potentially) reads from the environment or state and thus requires "view"
+		CHECK_WARNING_ALLOW_MULTI(
 			"contract C { function f() pure public { var x = " + x + "; x; } }",
```

Comment:
This is a tricky one. Perhaps you could turn this vector of expressions into a vector of pairs of expressions and their type, but it might not work for all of them because some types do not have a name.

Reply:
I took a stab at this but ran into some problems. Maybe you can spot the root cause? [gist](https://gist.github.com/jevogel/7371648e77dc7157a0e0e00716f80ea9).